src/components/TimeCardPage/CellTypes/CommentModals/ShowReportModal.tsx

```diff
@@ -142,73 +179,99 @@ export default function ShowReportModal({
   }
 
   const AddReportModal = () => {
-    const [remark, setRemark] = useState<ReportOptions>(ReportOptions.Late);
-    const [submitDisabled, setSubmitDisabled] =  useState(true);
-    const [selectedTime, setSelectedTime] = useState(moment(date)); // fix this rn
-    const user = useContext(UserContext);
-    
+    const [submitDisabled, setSubmitDisabled] =  useState(false);
 
-    const handleRemarkChange = (e) => {
-      setRemark(e.target.value);
-      
-      if (e.target.value === ReportOptions.Absent) {
-        setSubmitDisabled(false);
-      }
-    };
+    const [reason, setReason] = useState<ReportOptions>(ReportOptions.Late);
+    const [notify, setNotify] = useState('Yes');
+    const [explanation, setExplanation] = useState('');
 
-    const handleTimeChange = (e) => {
-      if (e.target.value) {
-        setSubmitDisabled(false);
-        setSelectedTime(selectedTime.hour(e.target.value.split(":")[0]))
-        setSelectedTime(selectedTime.minute(e.target.value.split(":")[1]))
-      }
-    };
+    const user = useContext(UserContext);
+    const toast = useToast();
 
-    const handleSubmit = () => {
-      if (reports.filter(report => report.Content === remark).length === 0) {
-        setReports([...reports, createNewReport(user, remark, selectedTime.unix())]);
-        console.log(createNewReport(user, remark, parseInt(selectedTime.format('X')))); // currently gmt TODO: fix later
-      }
+    const handleReasonChange = (option) => {
+      setReason(option as ReportOptions);
+    }
 
-      // show a toast that its been submitted
+    const handleSubmit = (e) => {
+      e.preventDefault();
+      if (reports.filter(report => report.Content === reason).length === 0) {
+        setReports([...reports, createNewReport(user, reason, notify, explanation)]);
```

Comment:
This is for my own clarification, but does the `setReports` function update the timesheet and send that update request to the API? 

Just double check that it will throw an error if something actually goes wrong - otherwise, the failed toast notif will never show.